frontend/pages/shopping-lists/_id.vue

```diff
@@ -656,7 +667,8 @@ export default defineComponent({
       loadingCounter.value += 1;
 
       // make sure it's inserted into the end of the list, which may have been updated
-      createListItemData.value.position = shoppingList.value?.listItems?.length || 1;
+      createListItemData.value.position = shoppingList.value?.listItems?.length
+        ? Math.max(...shoppingList.value.listItems.map(item => item.position || 0)) + 1 : 0;
```

Comment:
Could we use a reduce here? it would avoid creating a whole new array?

Reply:
Ope, definitely. TIL about reduce